### ISSUE TITLE
processing the dataset.

### ISSUE THREAD

**Issue by Esmail-ibraheem** [Lightning-AI/litgpt#1549]
I have an Arabic dataset of size 96GB that I want to use for pre-training litGPT. However, in the image provided [link to the image], it is mentioned that if the dataset is large, we should use litdata. But when I checked the README of litdata, there were no clear instructions on how to do it.

![big_data](https://github.com/Lightning-AI/litgpt/assets/113830751/c64e4382-40f7-4e7d-8cf5-ddf0f1ecb01f)

Here is the dataset I want to use: https://huggingface.co/datasets/ClusterlabAi/101_billion_arabic_words_dataset

Thank you.

**Comment by rasbt** [comment]
Good point. Does the LitData section here help? https://github.com/Lightning-AI/litdata?tab=readme-ov-file#1-prepare-your-data

**Comment by Esmail-ibraheem** [comment]
no, I did not understand from the litdata, how I can convert or process my custom [dataset](https://huggingface.co/datasets/ClusterlabAi/101_billion_arabic_words_dataset) so I can use it in litgpt:

**Comment by rasbt** [comment]
Personally, I use the `TextFiles` approach that I've implemented in LitGPT. But going back to an earlier comment you had, (and the phrase in the docs), my colleagues don't recommend it for very large datasets since it starts with plain text files (rather than tokenized text), and plain text can be inefficient to store. 

Personally, I don't have much experience with LitData, but If I ever prepare a large custom dataset, I'll amend the docs. In the meantime, the best way is perhaps to look at how its done for the `prepare_slimpajama.py` and `prepare_starcoder.py` in https://github.com/Lightning-AI/litgpt/tree/main/litgpt/data

which are used in the [Pretrain TinyLlama](https://github.com/Lightning-AI/litgpt/blob/main/tutorials/pretrain_tinyllama.md). Thomas Chaton, who is the developer of LitData, also has a tutorial on the dataset prep [here](https://lightning.ai/lightning-ai/studios/prepare-the-tinyllama-1t-token-dataset) which could be helpful
